src/main/java/net/mcreator/ui/wysiwyg/WYSIWYGEditor.java

```diff
@@ -81,6 +82,7 @@ public class WYSIWYGEditor extends JPanel {
 	private final JButton removeComponent = new JButton(UIRES.get("18px.remove"));
 	private final JButton moveComponentUp = new JButton(UIRES.get("18px.up"));
 	private final JButton moveComponentDown = new JButton(UIRES.get("18px.down"));
+	private final JButton lockComponent = new JButton(UIRES.get("16px.lock"));

```

Comment:
We use 18px SVG icons here. The currently selected icon does not fit well with the UI. You need to make new SVG icon that fits this sizing here, or wait for someone from the team to assist you with drawing when they have time.

@Mysterious-Dev could help, or maybe me when I get some extra time slot

Reply:
If you use an external icon, make sure it fits our scaling of "18px" size icons, and that its license permits use with GPL-3.0 license and does not require special attribution or license mention. As this is quite hard to find, we usually draw icons ourselves.

Reply:
This one could be suitable since it is CC0 licensed: https://www.svgrepo.com/svg/136946/lock

Just make sure to match size parameter to 18px and also padding / active icon area